docs/rancher.md

```diff
@@ -0,0 +1,29 @@
+# Deploying Antrea on Rancher Managed Cluster
```

Comment:
I still have the same comment as last time. I believe it would make more sense to move all this information to the Rancher section in docs/kubernetes-installers.md. I am not sure this warrants a standalone document.